raster-foundry/raster-foundry
# Add frontend/import support for Planet imagery

## Overview

This commits continues work to fully integrate Planet imagery into Raster Foundry to make it easier to import data into Raster Foundry and run analyses.

The workflow for importing Planet imagery into Raster Foundry following this commit is the following:
 - Browse imagery on Planet Explorer
 - Use API breakout to copy planet image IDs
 - Go to import workflow (add to existing project or a new one)
 - Choose *Planet*
 - Add API key if not present
 - Paste in Planet IDs

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
*Find imagery in Planet UI*
![planet-ui](https://user-images.githubusercontent.com/898060/29006726-c53636ce-7ac3-11e7-8b3e-71c93e7cd38c.png)

*Paste in Planet IDs*
![paste-planet-ids](https://user-images.githubusercontent.com/898060/29006731-cd5890d6-7ac3-11e7-9085-fb5bc71cef9f.png)

*Import running*
![dag-running](https://user-images.githubusercontent.com/898060/29006733-cf91ec26-7ac3-11e7-93a7-ba8761458bdf.png)

*Finished importing*
![imported-successfully](https://user-images.githubusercontent.com/898060/29006785-98875a62-7ac4-11e7-8f11-912b7317f3a8.png)

### Notes

In the future users will be able to browse Planet imagery directly in Raster Foundry; however, this is an intermediate step to make it easier to support some existing user workflows and use cases in the short-term.

## Testing Instructions

 * Start server with airflow, ensure all migrations are run
 * Add Rapid Eye + PlanetScope Datasources
 * Add API token for Planet in UI
 * Browse imagery in Planet explorer and find a few scenes to import (using the open California data is probably easiest)
 * Go through project create workflow and paste Planet Scene IDs when prompted
 * Grab a coffee/water/snack
 * Verify imagery imported successfully
